cardano-diffusion/ping/Cardano/Network/Ping.hs

```diff
@@ -0,0 +1,661 @@
+                     peerSharing   = PeerSharingDisabled,
+                     query
+                   }
+                   (const ())
+               )
+               [minBound..maxBound]
+
+           NodeToClient networkMagic query  ->
+             foldMapVersions
+               (\versionNumber ->
+                 simpleSingletonVersions
+                   versionNumber
+                   NodeToClientVersionData {
+                     networkMagic,
+                     query
+                   }
+                   (const ())
+               )
+               [minBound..maxBound]
+        )
+      case r of
+        Left err -> throwIO (PingClientProtocolLimitFailure err)
+        Right (Left err', rtt) -> do
+          logMsg (HandshakeRTT rtt)
+          throwIO (PingClientHandshakeProtocolError err' peerName)
+        Right (Right r', rtt) -> do
+          logMsg (HandshakeRTT rtt)
+          case r' of
+            HandshakeQueryResult versions -> do
+              -- print query results if it was supported by the remote side
+              unless pingOptsHandshakeQuery $
+                logMsg $ QueriedVersions (Map.keys versions)
+            HandshakeNegotiationResult _ version _versionData -> do
+              -- show negotiated version
+              logMsg $ NegotiatedVersion version
+              case protocol of
+                _ | pingOptsGetTip -> do
+                  --
+                  -- run chain sync to get the tip
+                  --
+                  let host =
+                        case Socket.addrAddress peer of
+                           Socket.SockAddrUnix path       -> Left path
+                           Socket.SockAddrInet pn ha      -> Right (IPv4 (fromHostAddress ha), pn)
+                           Socket.SockAddrInet6 pn _ ha _ -> Right (IPv6 (fromHostAddress6 ha), pn)
+                  stdGen <- initStdGen
+                  mx <- Mx.new
+                          Mx.nullTracers
+                          [MiniProtocolInfo {
+                            miniProtocolNum        = case protocol of
+                                                       NodeToNode {} -> NodeToNode.chainSyncMiniProtocolNum
+                                                       NodeToClient {} -> NodeToClient.localChainSyncMiniProtocolNum,
+                            miniProtocolDir        = Mx.InitiatorDirectionOnly,
+                            miniProtocolLimits     = case protocol of
+                                                       NodeToNode {} -> NodeToNode.chainSyncProtocolLimits NodeToNode.defaultMiniProtocolParameters
+                                                       NodeToClient {} -> NodeToClient.maximumMiniProtocolLimits,
+                            miniProtocolCapability = Nothing
+                          }]
+                  race_ (Mx.run mx bearer)
+                        (Mx.runMiniProtocol mx
+                          NodeToNode.chainSyncMiniProtocolNum
```

Comment:
It looks like you're using the wrong miniprotocol number here when running over NodeToClient.